### PR TITLE
API: verifyTealSign utility function

### PR DESCRIPTION
**Pull request by M-Picco**
Introduce a new function `verifyTealSign` which can be used to verify the output of its pair method, `tealSign`.

Some applications might benefit from having this utility function, either for testing purposes or not, and it would save developers the work of importing the `nacl` library and doing the verification manually by themselves.

### CODE REVIEW

File: src/logicsig.ts [hunk 1]
```diff
@@ -495,6 +495,29 @@ export function tealSign(
   return nacl.sign(toBeSigned, sk);
 }
 
+/**
+ * verifyTealSign verifies a signature as would the ed25519verify opcode
+ * @param sk - uint8array with public key to verify against
```
Comment (barnjamin) [inline src/logicsig.ts:500]: I think this should be `pk` and at the bottom?

Comment (M-Picco) [reply src/logicsig.ts:500]: You are right. I'll fix it

**Comment by bbroder-algo** [comment]
I don't have any problem with this nacl helper function, and LGTM, but I am not particularly qualified for JS.

**Review by jasonpaulos — APPROVED** [review]
This is an appreciated change. I believe we intended to add this at one point, but we never got around to it. Implementation and tests look good, as long as it passed CI.

**Pull request merged**